### PR TITLE
RavenDB-14093: add additional information to describe smaps reader error

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Sparrow.Server/Platform/Posix/SmapsReader.cs [hunk 1]
```diff
@@ -292,7 +292,13 @@ private int ReadFromFile(Stream fileStream, int bufferIndex)
                             if (foundValue == false)
                                 ThrowNotContainsValidValue(term, currentProcess.Id);
                             if (foundK == false)
-                                ThrowNotContainsKbValue(term, currentProcess.Id);
+                            {
+                                var additionalInfo =
+                                    $"Additional Info: switchBuffer={switchBuffer}, foundK/B={foundK}/{foundValue}, valueSearchPosition={valueSearchPosition}, bytesSearched={bytesSearched}" +
+                                    $", posInTempBuf={posInTempBuf}, searchedBuffer={searchedBuffer}, _tempBufferBytes=<{Encoding.UTF8.GetString(_tempBufferBytes)}" +
+                                    $", buffer 0=<{Encoding.UTF8.GetString(_smapsBuffer[0])}>. buffer 1={Encoding.UTF8.GetString(_smapsBuffer[1])}End of Addtional Info.";
```
Comment (ayende) [inline src/Sparrow.Server/Platform/Posix/SmapsReader.cs:299]: `Encoding.UTF8.GetString(_smapsBuffer[0])}`  - This is going to get the *whole* buffer, but we didn't read all of it, did we?

Comment (aviviadi) [reply src/Sparrow.Server/Platform/Posix/SmapsReader.cs:299]: true, but I want to _see_ the entire buffer. I have the positions where we read and from the text we can understand what went wrong.

**Comment by ppekrol** [comment]
@aviviadi Have you seen those failures? https://jenkins.hibernatingrhinos.com/job/PR_Tests_v4.2/2281/testReport/ 
Any idea?

**Comment by aviviadi** [comment]
> 
> 
> @aviviadi Have you seen those failures? https://jenkins.hibernatingrhinos.com/job/PR_Tests_v4.2/2281/testReport/
> Any idea?

I believe this is related to : RavenDB-13990, not here..

**Comment by ppekrol** [comment]
Let's rebase this PR then.

**Comment by ppekrol** [comment]
Rebased

**Pull request merged**